### ISSUE TITLE
A lot of useless space for a bit of useful info?

### ISSUE THREAD

**Issue by ArtySh0ck** [bvschaik/julius#281]
![image](https://user-images.githubusercontent.com/46932541/73175910-e592ab00-4113-11ea-9555-364659a1ba03.png)

Maybe this vertical mural is valued as a historical value, but very useless, u know.
But, behind, there is a good place to show informations? Unemployment % for example? Current Taxes\Wages\Taxing Area? Current people happiness? Count of Luxury Palaces for now? Count of free houses to immigrate?

**Comment by KillahNeo** [comment]
I've suggested tweaking that aswell, but i got an response that it would take way too much time to work on that plus maybe it couldn't even be done at all

**Comment by bvschaik** [comment]
Can we perhaps add the game speed and ratings here, instead of in the top bar?

**Comment by ArtySh0ck** [comment]
Maybe...

**Comment by crudelios** [comment]
Humm.. On the top bar the default green font looks good because the background image is more or less an uniform color.

That's not the case here, so some background would have to be added in order for the text to be readable.

For debugging purposes I sometimes use that part of the sidebar and I basically give it the same background of a tooltip: a black, 1 pixel border and then a white rectangle with the text. That is good enough for debugging purposes, but I think it looks ugly for an effective UI element.

So my question is: what could be used instead?

**Comment by bvschaik** [comment]
The same background as used for the "inner" panels in the advisors/building info?

![Screenshot 2020-01-28 at 18 22 50](https://user-images.githubusercontent.com/104874/73288425-40eb9880-41fb-11ea-9cfd-baff5402187d.png)

**Comment by crudelios** [comment]
Looks good! Is that a "programmer's mockup"? ;)

**Comment by ArtySh0ck** [comment]
Wow, it looks very good.

**Comment by bvschaik** [comment]
> Looks good! Is that a "programmer's mockup"? ;)

Yes, a one-minute "just draw an inner panel here please" programmer's mockup 😉 

So, next question: what do we want to have displayed in this panel?

**Comment by crudelios** [comment]
Game speed (with buttons)?
Expanded ratings?

```
Culture x/total
Prosperity x/total
Peace x/total
Favor x/total
Target population is X
```

I'm not sure the target population will fit.

**Comment by ArtySh0ck** [comment]
Happiness of people, unemployment %...

**Comment by bvschaik** [comment]
![game-speed](https://user-images.githubusercontent.com/104874/73354887-13ecc380-4297-11ea-8e61-f423812ce596.png)

So far I have this. I tried adding the ratings but there's not enough space for "prosperity 80/100" on one line, and I didn't even try more verbose languages (German, I'm looking at you).

**Comment by crudelios** [comment]
> So far I have this.

I like it!

For the rating, how about `Objectives 5/5` and the specofic ratings appear on a tooltop?

**Comment by m-mohr** [comment]
Re missing space: Maybe put it into two lines like with game speed?

Culture
x/total

Prosperity
x/total

Peace
x/total

Favor
x/total

**Comment by 411752230** [comment]
> ![game-speed](https://user-images.githubusercontent.com/104874/73354887-13ecc380-4297-11ea-8e61-f423812ce596.png)
> 
> So far I have this. I tried adding the ratings but there's not enough space for "prosperity 80/100" on one line, and I didn't even try more verbose languages (German, I'm looking at you).

This one is good for android touching supprt!

**Comment by m-mohr** [comment]
> I didn't even try more verbose languages (German, I'm looking at you).

@bvschaik As a German, I'm aware of the issue, but in this case it's quite funny that German is interestingly **less** verbose in most cases:

`Culture` (7 chars)
`Kultur` (6 chars)

`Prosperity` (10)
`Wohlstand` (9)

`Peace` (5)
`Frieden` (7)

`Favor` (5)
`Gunst` (5)

 :-)

**Comment by crudelios** [comment]
> Although on phones this is probably cut off (at least on my phone).

It most probably won't fit on a phone. It will fit in tablets though.

**Comment by ArtySh0ck** [comment]
> there's not enough space for "prosperity 80/100" on one line

maybe prosp. for prosperity? cult. for culture? something like that

**Comment by bvschaik** [comment]
> @bvschaik As a German, I'm aware of the issue, but in this case it's quite funny that German is interestingly **less** verbose in most cases:

Haha, irony indeed 🤣 

> maybe prosp. for prosperity? cult. for culture? something like that

That won't work. How do you abbreviate it in Russian or Chinese?

> It most probably won't fit on a phone. It will fit in tablets though.

It also fits (barely) on Switch and Vita, which both use a resolution of 960x540.

**Comment by bvschaik** [comment]
First test build is available here: https://bintray.com/bvschaik/julius-branches/windows/sidebar-extra-info-1.2.0.54-0ea8d24f#files

**Comment by crudelios** [comment]
Works fine for me!

**Comment by m-mohr** [comment]
Works for me, too.

**Comment by Legionary2** [comment]
Works for me!

**Comment by 411752230** [comment]
Works flawlessly with Chinese

**Comment by KillahNeo** [comment]
Refreshing to see UI changes being done

**Comment by ArtySh0ck** [comment]
Guys, how may i install another language of c3 but no corrupt my english c3? i want to try russian version with this test build. what i have to do?

**Comment by crudelios** [comment]
> Guys, how may i install another language of c3 but no corrupt my english c3? i want to try russian version with this test build. what i have to do?

You will need some extra image files besides the language files, do you have them?

If you do, I usually do the following to change the language:

1. Change the extension of all `*.eng` files to something like `*.engold`
2. Copy the new language files as the `*.eng` files
3. Play the game

To revert back to english, I usually do this:

1. Change the extension of the current language to its idiom. For example, if you have the russian files as the `*.eng`, I change them all to `*.rus`
2. Revert back the `*.engold` to `*.eng`

As an alternative, you can set a folder for each language. It's your choice, really. What matters is that the `*.eng` files correspond to the language you want.

**Comment by ArtySh0ck** [comment]
I just did another installation of 1C rus version ))
And for russian version this sidebar looks good too!
![image](https://user-images.githubusercontent.com/46932541/73448327-42d36a00-4369-11ea-9b7c-754e061d0344.png)

**Comment by ArtySh0ck** [comment]
and it disappeared... why 1.2.0.56 doesnt have a speed game bar ? =(

**Comment by crudelios** [comment]
> why 1.2.0.56 doesnt have a speed game bar ? =(

Because it's a work in progress that's dependent on #284

**Comment by bvschaik** [comment]
The latest master now has this information (if window is sufficiently tall):

<img width="191" alt="sidebar-info-v2" src="https://user-images.githubusercontent.com/104874/73616350-86043600-4612-11ea-89da-c25bc290d84d.png">

In case of worker shortage, this will display "0% (-15)" if you are 15 workers short.

**Comment by bvschaik** [comment]
I just had a light-bulb moment for the ratings. How about this? Green = reached, red = not yet reached.

<img width="181" alt="sidebar-info-ratings" src="https://user-images.githubusercontent.com/104874/73872715-838f1f80-4850-11ea-8773-5b9711a7e0fc.png">

**Comment by crudelios** [comment]
Are those different colors? I'm color blind (I have difficulty precisely in the red-green spectrum) and can't notice the difference...

Also, most times people want to know the rating objective.

What if you simply reduce the font size?

**Comment by m-mohr** [comment]
@crudelios Yes, the "58" is red.

@bvschaik It would still be useful to get the required numbers. If there's not enough space, maybe using tooltips? That could also be added as tooltip to the population count in the top menu.

Another light-bulb moment: My idea is to do it similarly as for the unemployment. Show the difference from the required numbers. So If Peace required 70 and you have 58, show -12. If Culture required 50 and you have 90, show +40. That would solve the color issue and basically shows you what to do without showing two numbers.

**Comment by bvschaik** [comment]
@m-mohr Showing the difference will be very confusing. I'm definitely not thinking "oh, I need 3 more culture rating and then I'm done".

Ok, how about this, then?
<img width="188" alt="sidebar-info-ratings-v2" src="https://user-images.githubusercontent.com/104874/73958163-ceb83980-4907-11ea-965f-0ff211d14b0b.png">

**Comment by m-mohr** [comment]
@bvschaik I don't think it is confusing, but maybe I'm just trained to think differently in this regard ;-)

I like the implementation shown, although the population number is then shown twice on the screen.

And you are saying German is verbose? Italian is! :-D

**Comment by bvschaik** [comment]
Population is now indeed shown twice. The alternative is not having it there but then you miss the requirement.

French can also be very verbose for some phrases.

**Comment by crudelios** [comment]
I really like that last implementation, even with the repeated population. However, can you make the font colors for the achieved and not achieved objectives a bit more different please?

**Comment by bvschaik** [comment]
Unfortunately not. These are the prerendered, anti-aliased, and shadowed fonts `FONT_NORMAL_GREEN` and `FONT_NORMAL_RED`, which cannot be colored. Even if they could, I don't want to mess with the colors there since any tinting easily look wrong.

**Comment by crudelios** [comment]
Oh it's those fonts. Ok, no problem then!

**Comment by bvschaik** [comment]
Alright. Latest unstable has the ratings added.
I'm going to close this issue, if anyone has more ideas for things to add, please open a new issue.
